querybuilder/rule.go

```diff
@@ -72,41 +83,39 @@ func (r *Rule) getInputValue(dataset map[string]interface{}) interface{} {
 		} else if rresult.Kind() == reflect.Slice && i != (steps-1) {
 			result = rresult.Index(0)
 		}
-
-		if result == nil {
-			return nil
-		}
 	}
 
-	iv := r.parseValue(result)
+	iv, err := r.parseValue(result)
+
 	if r.Sanitize && r.Type == "string" {
 		v := iv.(string)
-		return sanitize(&v)
+		return sanitize(&v), err
 	}
 
-	return iv
+	return iv, err
```

Comment:
why are we returning errors here?

Reply:
This function `getInputValue()` is called in `Evaluate()` function in `rule.go`, so we need to propagate back the error message.